keylime/tpm_abstract.py

```diff
@@ -337,12 +346,17 @@ def __check_ima(self, agent_id, pcrval, ima_measurement_list, ima_whitelist):
         return True
 
     def check_pcrs(self, agent_id, tpm_policy, pcrs, data, virtual, ima_measurement_list, ima_whitelist):
+        try:
+            tpm_policy = ast.literal_eval(tpm_policy)
+        except ValueError:
+            pass
```

Comment:
If the ValueError is caught here, the next lin will raise a NameError. It's better to raise or provide a default.

Reply:
Sorry, my bad. `tpm_policy` is already defined in the function signature.

Reply:
However, I would go with another name instead of reusing it. The problem is that `ast.literal_eval()` will rehydrate the string to whatever type it will evaluate to. If the exception is caught then `tpm_policy` will remain a `str` which does not have a `copy()` method. I'd go with:
```
try:
    tpm_policy = ast.literal_eval(tpm_policy)
except ValueError:
    tpm_policy_ = {} # If you expect it to be a dict
pcrWhiteList = tpm_policy_.copy()